components/SideMenu.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <nav v-sticky="shouldStick" sticky-offset="offset" sticky-side="top" flex column class="sidebar">
+    <a v-for="item in items" :key="item.id" :href="`#${item.id}`">{{item.text}}</a>
+  </nav>
+</template>
+
+<script>
+export default {
+  data () {
```

Comment:
this could be move in `props`.

Also I like to keep the order
- props
- data
- computed
- methods
- hooks (mounted, beforeDestroy)

I feel it makes more sense for dependencies (data depends on props, computed depends on data, methods depends on computed and data, hooks depends on methods)

If there is any guidelines for that i'm happy to follow them, if not we should keep it consistent with the rest of the codebase

Reply:
I really like your guideline for ordering :)